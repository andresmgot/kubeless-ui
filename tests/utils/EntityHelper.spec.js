/*
Copyright 2017 Bitnami.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/
import EntityHelper from 'utils/EntityHelper'

describe('(Utils) EntityHelper', () => {

  it('updateEntityInList - should update entity in List based on UID', () => {
    const list = [
      { metadata: { uid: 'uid1', name: 'A' } },
      { metadata: { uid: 'uid2', name: 'B' } }
    ]
    const updatedEntity = { metadata: { uid: 'uid2', name: 'B2' } }
    expect(EntityHelper.updateEntityInList(list, updatedEntity)).toMatchSnapshot()
  })

  it('deleteEntityInList - should delete entity in List based on UID', () => {
    const list = [
      { metadata: { uid: 'uid1', name: 'A' } },
      { metadata: { uid: 'uid2', name: 'B' } }
    ]
    const deleteEntity = { metadata: { uid: 'uid2', name: 'B' } }
    expect(EntityHelper.deleteEntityInList(list, deleteEntity)).toMatchSnapshot()
  })

  it('functionFromParams - should create Function entity from params', () => {
    const params = {
      name: 'funcName',
      namespace: 'kubeless',
      'function': '//code goes here',
      handler: 'funcName.foo'
    }
    expect(EntityHelper.functionFromParams(params)).toMatchSnapshot()
  })

  it('entityStatus - should return status of Pod', () => {
    const podRunning = {
      metadata: {},
      status: { phase: 'Running' }
    }
    expect(EntityHelper.entityStatus(podRunning)).toEqual('Running')
    const podTerminating = {
      metadata: { deletionTimestamp: new Date() },
      status: { phase: 'Running' }
    }
    expect(EntityHelper.entityStatus(podTerminating)).toEqual('Terminating')
    const podContainerCreating = {
      metadata: {},
      status: { containerStatuses: [{
        state: {
          waiting: { reason: 'ContainerCreating' }
        }
      }] }
    }
    expect(EntityHelper.entityStatus(podContainerCreating)).toEqual('ContainerCreating')
  })

})
