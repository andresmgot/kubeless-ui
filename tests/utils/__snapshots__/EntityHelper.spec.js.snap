// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Utils) EntityHelper deleteEntityInList - should delete entity in List based on UID 1`] = `
Array [
  Object {
    "metadata": Object {
      "name": "A",
      "uid": "uid1",
    },
  },
]
`;

exports[`(Utils) EntityHelper functionFromParams - should create Function entity from params 1`] = `
Object {
  "apiVersion": "kubeless.io/v1beta1",
  "kind": "Function",
  "metadata": Object {
    "name": "funcName",
    "namespace": "kubeless",
  },
  "spec": Object {
    "checksum": "sha256:e4032e36ebd8b1e2e7249bff8ef675a32d01a1f96ead2b6979fa9d3a2c53f371",
    "deps": "",
    "function": "//code goes here",
    "handler": "funcName.foo",
    "runtime": undefined,
  },
}
`;

exports[`(Utils) EntityHelper updateEntityInList - should update entity in List based on UID 1`] = `
Array [
  Object {
    "metadata": Object {
      "name": "A",
      "uid": "uid1",
    },
  },
  Object {
    "metadata": Object {
      "name": "B2",
      "uid": "uid2",
    },
  },
]
`;
